src/python_testing/TC_PWRTL_2_1.py

```diff
@@ -40,36 +40,43 @@ async def test_TC_PWRTL_2_1(self):
-            return
+        self.print_step(1, "Commissioning, already done")
 
         self.print_step(2, "Read AvailableAttributes attribute")
-        available_endpoints = await self.read_single_attribute_check_success(endpoint=endpoint, cluster=Clusters.Objects.PowerTopology, attribute=attributes.AvailableEndpoints)
-
-        if available_endpoints == NullValue:
-            logging.info("AvailableEndpoints is null")
-        else:
-            logging.info("AvailableEndpoints: %s" % (available_endpoints))
+        if avail_endpoints_attr_id in attribute_list:
+            available_endpoints = await self.read_single_attribute_check_success(endpoint=endpoint, cluster=Clusters.Objects.PowerTopology, attribute=attributes.AvailableEndpoints)
 
-            asserts.assert_less_equal(len(available_endpoints), 21,
-                                      "AvailableEndpoints length %d must be less than 21!" % len(available_endpoints))
+            if available_endpoints == NullValue or available_endpoints == []:
+                logging.info("AvailableEndpoints is null or an empty list")
+            else:
+                logging.info("AvailableEndpoints: %s" % (available_endpoints))
+                asserts.assert_less_equal(len(available_endpoints), 21,
+                                          "AvailableEndpoints length %d must be less than 21!" % len(available_endpoints))
 
-        if not self.check_pics("PWRTL.S.A0001"):
-            logging.info("Test skipped because PICS PWRTL.S.A0001 is not set")
+        else:
+            logging.info('Skipping test as available endpoints attribute ID not in attribute list on DUT')
             return
 
         self.print_step(3, "Read ActiveEndpoints attribute")
-        active_endpoints = await self.read_single_attribute_check_success(endpoint=endpoint, cluster=Clusters.Objects.PowerTopology,  attribute=attributes.ActiveEndpoints)
-        logging.info("ActiveEndpoints: %s" % (active_endpoints))
+        if act_endpoints_attr_id in attribute_list:
+            active_endpoints = await self.read_single_attribute_check_success(endpoint=endpoint, cluster=Clusters.Objects.PowerTopology,  attribute=attributes.ActiveEndpoints)
+            logging.info("ActiveEndpoints: %s" % (active_endpoints))
+
+            if available_endpoints == NullValue or available_endpoints == []:
+                asserts.assert_true(active_endpoints == NullValue or active_endpoints == [],
+                                    "ActiveEndpoints should be null when AvailableEndpoints is null: %s" % active_endpoints)
 
-        if available_endpoints == NullValue:
-            asserts.assert_true(active_endpoints == NullValue,
-                                "ActiveEndpoints should be null when AvailableEndpoints is null: %s" % active_endpoints)
+        else:
+            logging.info('Skipping test as active endpoints attribute ID not in attribute list on DUT')
```

Comment:
can you change this to say that it's the test step being skipped?

Reply:
Hi Cecille,
I have updated to show that test step 3 has been skipped